### PR TITLE
Added ColumnValues to Row-level Results

### PR DESCRIPTION
**Pull request by zixianzh1**
*Issue #, if available:*

*Description of changes:*
Add Max, Min, PatternMatch to rowlevel results
Change the null behavior back

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/scala/com/amazon/deequ/constraints/Constraint.scala [hunk 1]
```diff
@@ -316,7 +316,13 @@ object Constraint {
     val constraint = AnalysisBasedConstraint[NumMatchesAndCount, Double, Double](
       compliance, assertion, hint = hint)
 
-    new NamedConstraint(constraint, s"ComplianceConstraint($compliance)")
+//    new NamedConstraint(constraint, s"ComplianceConstraint($compliance)")
```
Comment (mentekid) [inline src/main/scala/com/amazon/deequ/constraints/Constraint.scala:319]: nit: remove commented out code

File: src/test/scala/com/amazon/deequ/analyzers/MaximumTest.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import com.amazon.deequ.utils.FixtureSupport
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+class MaximumTest extends AnyWordSpec with Matchers with SparkContextSpec with FixtureSupport {
+
+  "Max" should {
+    "return row-level results for columns" in withSparkSession { session =>
+
+      val data = getDfWithNumericValues(session)
```
Comment (mentekid) [inline src/test/scala/com/amazon/deequ/analyzers/MaximumTest.scala:32]: can you add a null to this frame?

File: src/test/scala/com/amazon/deequ/analyzers/ComplianceTest.scala [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import com.amazon.deequ.utils.FixtureSupport
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+class ComplianceTest extends AnyWordSpec with Matchers with SparkContextSpec with FixtureSupport {
+
+  "Compliance" should {
+    "return row-level results for columns" in withSparkSession { session =>
+
+      val data = getDfWithNumericValues(session)
```
Comment (mentekid) [inline src/test/scala/com/amazon/deequ/analyzers/ComplianceTest.scala:32]: can you add a null to this frame to show how this works for nulls?

File: src/test/scala/com/amazon/deequ/analyzers/MinimumTest.scala [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import com.amazon.deequ.utils.FixtureSupport
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+class MinimumTest extends AnyWordSpec with Matchers with SparkContextSpec with FixtureSupport {
+
+  "Min" should {
+    "return row-level results for columns" in withSparkSession { session =>
+
+      val data = getDfWithNumericValues(session)
```
Comment (mentekid) [inline src/test/scala/com/amazon/deequ/analyzers/MinimumTest.scala:32]: same here for nulls

File: src/test/scala/com/amazon/deequ/analyzers/PatternMatchTest.scala [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import com.amazon.deequ.utils.FixtureSupport
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+class PatternMatchTest extends AnyWordSpec with Matchers with SparkContextSpec with FixtureSupport {
+
+  "PatternMatch" should {
+    "return row-level results for non-null columns" in withSparkSession { session =>
+
+      val data = getDfWithStringColumns(session)
```
Comment (mentekid) [inline src/test/scala/com/amazon/deequ/analyzers/PatternMatchTest.scala:30]: same for all these for nulls

**Review by mentekid — APPROVED** [review]

**Comment by torontoairbnb** [comment]
hi @mentekid  I can see this PR mereged to master and PatternMatch and min max record level has been added, I have tested the verificationSuiteTest piece of code but only two rules result showing in thev rowLevelResultAsDataFrame which are for isComplete and completeness and not the new rules

**Pull request merged**